UtilsDynamicSimulations/OpenSimAD/utilsKineticsOpenSimAD.py

```diff
@@ -0,0 +1,420 @@
+            with columns named after the coordinates.
+    
+        Notes:
+            - Rotational degrees of freedom are reported in degrees per second
+              squared.
+            - Translational degrees of freedom are reported in meters per second
+              squared.
+        """
+        coordinate_accelerations = np.copy(
+            self.optimal_result['coordinate_accelerations_toTrack'])
+        coordinate_accelerations[self.idxColumnRotLabels, :] *= 180/np.pi
+    
+        output = pd.DataFrame(coordinate_accelerations.T, 
+                              columns=self.coordinate_names)
+        output.insert(0, 'time', self.time)
+    
+        return output
+    
+    def get_ground_reaction_forces(self):
+        """
+        Retrieves the ground reaction forces from the dynamic simulation.
+    
+        Returns:
+            pandas.DataFrame: A DataFrame containing ground reaction forces
+            with columns named after the side and direction.
+    
+        Notes:
+            - Forces are reported in Newtons.
+            - Side: right and left indicate right and left legs, respectively.
+            - Direction: x, y, and z indicate the posterior-anterior, 
+              inferior-superior, and medial-lateral directions, respectively.
+        """   
+        output = pd.DataFrame(self.optimal_result['GRF'].T, 
+                              columns=self.optimal_result['GRF_labels'])
+        output.insert(0, 'time', self.time)
+    
+        return output
+    
+    def get_ground_reaction_moments(self):
+        """
+        Retrieves the ground reaction moments from the dynamic simulation.
+    
+        Returns:
+            pandas.DataFrame: A DataFrame containing ground reaction moments
+            with columns named after the side and direction.
+    
+        Notes:
+            - Moments are reported in Newton-meters.
+            - The moments are expressed in the ground reference frame with 
+              respect to the ground origin.
+            - Side: right and left indicate right and left legs, respectively.
+            - Direction: x, y, and z indicate the posterior-anterior, 
+              inferior-superior, and medial-lateral directions, respectively.
+        """   
+        output = pd.DataFrame(self.optimal_result['GRM'].T, 
+                              columns=self.optimal_result['GRM_labels'])
+        output.insert(0, 'time', self.time)
+    
+        return output
+    
+    def get_ground_reaction_free_moments(self):
+        """
+        Retrieves the ground reaction free moments from the dynamic simulation.
+    
+        Returns:
+            pandas.DataFrame: A DataFrame containing ground reaction free 
+            moments with columns named after the side and direction.
+    
+        Notes:
+            - Moments are reported in Newton-meters.
+            - The moments are expressed in the ground reference frame with 
```

Comment:
free torque is expressed in ground frame, and since it is a torque, there's no "point" where it is expressed about. I have verified that it is 0 in all directions but y, which is correct.